added per world config and blacklist worlds

DeathGOD7 opened the pull request:
Well as the name suggests I added some per world config and blacklist world that I was talking about in #6 .
It still needs debug logs to be removed and have some testing with other plugins like essential.

The test is done with build in my forked repo...with Multiverse Core
It works but if you have to SMP worlds in the same server..and have a bed respawn in world A and you die in world B...then you are randomly tped in the world A cause you have bed respawn there....and the respawn event takes the world you are going to spawn in.

To fix that you just need to disable bed respawns in multiverse worlds.yml for World B. But I haven't checked whether if it spawns you near bed...when having bed respawn in "world B" worlds.yml to false.

DeathGOD7 committed:
Added per world config and blacklist world

Things that were added :
- global values for world that don't have custom config
- toggle switch for random respawn on death
- fixed the config keys so it doesn't make plugin users confused
- added per world config settings

To Do List :
- remove debuggers in code
- check for plugin error and bugs

Code Updated by : https://github.com/DeathGOD7

DeathGOD7 committed:
Update README.md

DeathGOD7 committed:
added global and per world respawn world

Added per world and global respawn world.

Updated :
1. Removed the some debugger in code (some are left like respawn world as it can be helpful or can be taken as normal log)
2. Added per world and global respawn-world. (so you can define where the player is respawned if they dies.)

Note : if you have multiple world for eg. survival and pvp...and you have bed respawn in survival and player dies in pvp they will respawn randomly in survival world. (even if you set respawn world to same world : pvp and disable random respawn on death)
cause the current respawn world is only useful with random respawn enabled on death.

to fix that you should disable bed respawn in multiverse world.yml and add that world(pvp) to blacklist

To do list
- add feature that if user custom respawn world and disables random respawn on death...that it will tp them to that respawn world spawn point.

DeathGOD7 committed:
removed the unused namespaces

removed unused namespaces that were added by mistake
and added test build of this changes

Shivelight commented:
I will review the code in a few hours. Thank you!

DeathGOD7 commented:
Update : 
- added per world respawn world
- removed all debug code..some are left to log the death as its useful for server

## Minor Bug
You will respawn in the spawn point of Nether world if you are using the old version of Multiverse 4.2.2 and below and running a 1.16+ server.
To fix that just use the latest build or v 4.3.0 or higher where multiverse fixed that issue.

DeathGOD7 committed:
update readme

updated readme file to let user know its not our bug but in multiverse side plugins when user tp to spawn point or 0 y 0 in nether.

Shivelight commented on pom.xml:
```diff
@@ -6,7 +6,7 @@
     <groupId>id.shivelight.paper</groupId>
     <artifactId>unexpectedspawn</artifactId>
-    <version>0.2.1</version>
+    <version>0.2.2</version>
```
This is huge and breaking changes, bump the version to 0.3.0.

RexSE7EN replied:
Will update the version once I am back on PC. Currently on phone.

Shivelight commented on src/main/java/id/shivelight/paper/unexpectedspawn/Spawn.java:
```diff
@@ -37,52 +38,191 @@
+        }
+    }
+
+    private void logMessageConsole(String msg, String type) {
+        String logPrefix = "[UnexpectedSpawn] ";
+        if (type.equals("severe")) {
+            Bukkit.getLogger().severe(logPrefix + msg);
+        }
+        else if (type.equals("warn")) {
+            Bukkit.getLogger().warning(logPrefix + msg);
+        }
+        else {
+            Bukkit.getLogger().info(logPrefix + msg);
```
`type` parameter can be an enum. Furthermore, it is a good idea to extract this method to a new class and reuse it.

RexSE7EN replied:
Will check it and try implementing it. Thank You for your feedback.

Shivelight replied:
Even better, create methods for each level, e.g. Logger#warning, Logger#info, etc.

RexSE7EN replied:
So normal class or enum class?
Ok per log method.

Shivelight replied:
I was thinking both, but per log level methods is much better?

RexSE7EN replied:
So for enum one java file and for logger methods one?
You can dm me in discord also.

RexSE7EN replied:
It's done. +resolved

Shivelight commented on src/main/java/id/shivelight/paper/unexpectedspawn/UnexpectedSpawn.java:
```diff
@@ -28,9 +28,11 @@
-        config = new ConfigAccessor(this, "config.yml");
-        config.getConfig().options().copyDefaults(true);
-        config.saveConfig();
+        this.config = new ConfigAccessor(this, "config.yml");
+        this.config.getConfig().options().copyDefaults(true);
+//        config.saveConfig();
+//        used this so the comment in config.yml is preserved.
+        this.config.saveDefaultConfig();
```
Just adding future concerns; if the config structure is changed, it will not merge the existing configuration (new keys will be missing). Looks good for now.

RexSE7EN replied:
Yes, user need to remove any previous build or config files.

Shivelight reviewed (changes requested):
Looks good, just needs some changes.

DeathGOD7 committed:
bump version to 0.3.0 and added requested changes feature

updated things:
- bumped to 0.3.0 as it was considered huge changes
- fixed multiverse core soft depend (as there was typo)
- fixed requested changes (added new class for logger : LogConsole.java)
- made readme file look good
- fixed some var naming style (like blackListedWorlds => blacklistedWorlds)
- renamed per world blacklist block holder variable name

Shivelight reviewed (approved):
LGTM. Leaving the debug comments until the code is confirmed bug-free. Thank you!
